File: cloud-iep-client/src/goals/Goal.ts
import type { Observation } from './Observation';

interface Objective {
  objectiveName: string;
  complete: boolean;
}

export interface Goal {
  id: string;
  goalName: string;
  goalDescription: string;
  category: string;
  beginDate: Date;
  endDate: Date;
  goalPercentage: number;
  studentId: string;
  objectives: Objective[];
  observations: Observation[];
}

File: api/CloudIEP.Data/Models/Goal.cs
using System;
using System.Collections.Generic;

namespace CloudIEP.Data.Models
{
    public class Goal : Entity
    {
        public string GoalName { get; set; }
        public string GoalDescription { get; set; }
        public string Category { get; set; }
        public DateTime BeginDate { get; set; }
        public DateTime EndDate { get; set; }
        public decimal GoalPercentage { get; set; }
        public List<Objective> Objectives { get; set; } = new List<Objective>();
        public List<Observation> Observations { get; set; } = new List<Observation>();
        public string StudentId { get; set; }
    }

    public class Objective
    {
        public string ObjectiveName { get; set; }
        public bool Complete { get; set; }
    }

    public class Observation
    {
        public DateTime ObservationDate { get; set; }
        public int SuccessCount { get; set; }
        public int TotalCount { get; set; }
    }
}
